### PR TITLE
Add latency metrics for each visibility message to ES 

### PR DESCRIPTION
**Pull request by vancexu**
I also contains a small refactor of archival tests

### CODE REVIEW

File: common/metrics/fake.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
```
Comment (wxing1292) [inline common/metrics/fake.go]: could you change this file name to noop.go?

File: service/worker/archiver/replay_metrics_client.go [hunk 2]
```diff
@@ -67,7 +67,7 @@ func (r *replayMetricsClient) AddCounter(scope int, counter int, delta int64) {
 // StartTimer starts a timer for the given metric name. Time will be recorded when stopwatch is stopped.
 func (r *replayMetricsClient) StartTimer(scope int, timer int) tally.Stopwatch {
 	if workflow.IsReplaying(r.ctx) {
-		return nopStopwatch()
+		return metrics.NopStopwatch()
```
Comment (wxing1292) [inline service/worker/archiver/replay_metrics_client.go]: noop?

File: service/worker/indexer/esProcessor.go [hunk 3]
```diff
@@ -64,6 +65,11 @@ type (
 		logger        log.Logger
 		metricsClient metrics.Client
 	}
+
+	kafkaMessageWithMetrics struct { // value of esProcessorImpl.mapToKafkaMsg
```
Comment (wxing1292) [inline service/worker/indexer/esProcessor.go]: you can also make this struct implement messaging.Message and make the implemented ack / nack call stopwatch.stop

File: common/metrics/fake.go (hunk 1 above)
Comment (vancexu) [reply common/metrics/fake.go]: renamed

File: service/worker/archiver/replay_metrics_client.go (hunk 2 above)
Comment (vancexu) [reply service/worker/archiver/replay_metrics_client.go]: FYI 

> In computer science, a NOP, no-op, or NOOP (pronounced "no op"; short for no operation) is an assembly language instruction, programming language statement, or computer protocol command that does nothing.

File: service/worker/indexer/esProcessor.go (hunk 3 above)
Comment (vancexu) [reply service/worker/indexer/esProcessor.go]: updated, I think your suggestion is cleaner

**Pull request merged**